src/acp/rule.ts

```diff
@@ -0,0 +1,493 @@
+  return addIri(policy, acp.noneOf, rule);
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Set the rules restrincting the scope of a given the [[Policy]]. If an agent
+ * requesting access to a resource is present in **any** of the required rules,
+ * they will not be granted access.
+ * @param policy The [[Policy]] to which the rule should be added.
+ * @param rules The rules the policy accepts.
+ * @returns A new [[Policy]] clone of the original one, with the optional rules replaced.
+ * @since Unreleased
+ */
+export function setForbiddenRuleOnPolicy(
+  policy: Policy,
+  rules: Rule[]
+): Policy {
+  if (rules.length === 0) {
+    return removeAll(policy, acp.noneOf);
+  }
+  let updatedPolicy = setIri(policy, acp.noneOf, rules[0]);
+  rules.slice(1).forEach((ruleToAdd: Rule) => {
+    updatedPolicy = addForbiddenRuleToPolicy(updatedPolicy, ruleToAdd);
+  });
+  return updatedPolicy;
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Get the [[Rule]]'s forbidden by the given [[Policy]]
+ * @param policy The [[policy]] from which the rules should be read.
+ * @returns A list of the forbidden [[Rule]]'s
+ * @since unreleased
+ */
+export function getForbiddenRuleOnPolicyAll(policy: Policy): UrlString[] {
+  return getIriAll(policy, acp.noneOf);
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Initialise a new empty [[SolidDataset]] to store [[Rule]]'s in.
+ */
+export const createRuleDataset = createSolidDataset;
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Initialise a new, empty [[Rule]].
+ *
+ * @param url URL that identifies this [[Rule]].
+ */
+export function createRule(url: Url | UrlString): Rule {
+  const stringUrl = internal_toIriString(url);
+  let ruleThing = createThing({ url: stringUrl });
+  ruleThing = setUrl(ruleThing, rdf.type, acp.Rule);
+  return ruleThing;
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Get the [[Rule]] with the given URL from an [[RuleDataset]].
+ *
+ * @param ruleResource The Resource that contains the given [[Rule]].
+ * @param url URL that identifies this [[Rule]].
+ * @returns The requested [[Rule]], if it exists, or `null` if it does not.
+ */
+export function getRule(
+  ruleResource: RuleDataset,
+  url: Url | UrlString
+): Policy | null {
```

Comment:
I reckon the returned type here is wrong? Should be `Rule`?

Reply:
Fixed in 59e7de6daf35fcf97bc5f343d88b19575ee6717d